File: Back/PasswordManager.Api/Dto/PasswordDto.cs
namespace PasswordManager.Api.Dto;

/// <summary>
/// Data Transfer Object for Password entity (output).
/// </summary>
public class PasswordDto
{
    public int PasswordId { get; set; }
    public string AccountName { get; set; }
    public string EncryptedPassword { get; set; }
    public int ApplicationId { get; set; }
    
    // Optional
    public ApplicationDto Application { get; set; }
}

File: Front/password-manager-front/src/app/dto/password-dto.ts
/**
 * Data Transfer Object for retrieving a password.
 */
export interface PasswordDto {
  passwordId: number;
  accountName: string;
  encryptedPassword: string;
  applicationId: number;
}

/**
 * Data Transfer Object for creating a new password.
 * The plain password is provided and will be encrypted by the service.
 */
export interface CreatePasswordDto {
  accountName: string;
  plainPassword: string;
  applicationId: number;
}
